Oracle/diagnosis_covid_patch.sql

```diff
@@ -0,0 +1,346 @@
+    when raw_dx_source like '%:UHC_DIAGNOSIS%' then 'BI' -- from pcornet_diag
+    when raw_dx_source like '%PAT_ENC_DX%' or raw_dx_source like '%PROBLEM_LIST%' then 'OD' -- Order/EHR
+    when raw_dx_source like '%MEDICAL_HX%' then 'OT' -- medical history -> other
+    end dx_origin,
+    case
+    when raw_dx_source like '%SECOND%' then 'S'
+    when raw_dx_source like '%PRIMARY%' then 'P'
+    end pdx,
+    case when enc_type in ('EI', 'IP') then
+      case
+        when raw_dx_source like '%ADMIT%' then 'Y'
+        when raw_dx_source like ':POA%' then 'Y'
+        else 'NI'
+      end
+    end dx_poa,
+    substr(f.raw_dx, 1, 50) raw_dx,
+    null raw_dx_type,
+    substr(f.raw_dx_source, 1, 50) raw_dx_source,
+    null raw_origdx,
+    null raw_pdx,
+    null raw_dx_poa
+FROM
+    (
+    select patient_num, encounter_num, start_date, provider_id
+          , listagg(modifier_cd, ',') within group (order by modifier_cd) raw_dx_source
+          , listagg(concept_cd, ',') within group (order by concept_cd) raw_dx
+     from covid_dx_facts
+     group by patient_num, encounter_num, start_date, provider_id 
+    ) f
+    join pcornet_cdm.encounter enc on enc.encounterid = to_char(f.encounter_num)
+-- where enc.enc_type in ('IP', 'EI')
+-- where enc.encounterid is null
+-- order by patid, admit_date, dx_date
+;
+-- 4,234 rows inserted.
+-- KUMC 6,657  rows inserted.
+commit;
+
+/*
+select * from (
+select enc_type, count(distinct patid) pat_qty, count(distinct encounterid) enc_qty, count(*) obs_qty
+from diagnosis_c19_no_orphan_enc
+group by enc_type
+)
+union all
+select 'tot', count(distinct patid) pat_qty, count(distinct encounterid) enc_qty, count(*) obs_qty
+from diagnosis_c19_no_orphan_enc
+;
+
+select * from (
+select enc_type, count(distinct patid) pat_qty, count(distinct encounterid) enc_qty, count(*) obs_qty
+from diagnosis_c19
+group by enc_type
+)
+union all
+select 'tot', count(distinct patid) pat_qty, count(distinct encounterid) enc_qty, count(*) obs_qty
+from diagnosis_c19
+;
+
+-- browse columns etc.
+select * from pcornet_cdm.diagnosis;
+
+select round(100 * 10 / 1788,2 )from dual;
+
+
+create table pcornet_cdm.diagnosis_w_c19 as
+select * from pcornet_cdm_c8r3.diagnosis
+union all
+select * from diagnosis_c19_no_orphan_enc
+;
+*/
+
+insert /*+ APPEND */ into pcornet_cdm.diagnosis
+select * from diagnosis_c19_no_orphan_enc
+;
+--20,322 rows inserted.
+commit;
+/*
+select count (distinct patid) from  pcornet_cdm.diagnosis
+where dx='U07.1';
+--KUMC 2956
+*/
+/**** */
+
+
+-- TODO
+-- from https://github.com/kumc-bmi/i2p-transform/blob/master/Oracle/pcornet_loader.sql#L12
+  merge into pcornet_cdm.diagnosis d
+  using pcornet_cdm.encounter e
+     on (d.encounterid = e.encounterid)
+  when matched then update set d.providerid = e.providerid;
+-- from https://github.com/kumc-bmi/i2p-transform/blob/master/Oracle/pcornet_loader.sql#L107
+
+ update pcornet_cdm.diagnosis
+ set pdx='NI'
+ where pdx='1';
```

Comment:
Do we have to run these two scripts? my understanding is these two commands will get executed in pcornet_loader.sql anyway. @lav-patel

Reply:
the rest of the code looks fine with me

Reply:
This code is not integrated to run with Luigi, and it has a separate Jenkins job (pcornet_cdm_covid_patches) ,so this script will run at the end of the COVID ETL, so these two stamens will be needed as long as we keep this patch as a separate job.